subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -366,9 +369,17 @@ static int job_update_accepted(struct mqtt_client *const client, uint32_t payloa
 		LOG_DBG("Start downloading firmware from %s/%s",
 			(char *)hostname, (char *)file_path);
 
-#if defined(CONFIG_AWS_FOTA_DOWNLOAD_SECURITY_TAG)
-		sec_tag = CONFIG_AWS_FOTA_DOWNLOAD_SECURITY_TAG;
+		/* If the protocol is https set the sec_tag. This instructs the download client
+		 * to use https. Fail if the protocol is https but no sec tag is configured.
+		 */
+		if (strncmp(protocol, "https", 5) == 0) {
+#if defined(CONFIG_AWS_FOTA_DOWNLOAD_SECURITY_TAG) && CONFIG_AWS_FOTA_DOWNLOAD_SECURITY_TAG != -1
```

Comment:
```suggestion
if (IS_ENABLED(CONFIG_AWS_FOTA_DOWNLOAD_SECURITY_TAG) && (CONFIG_AWS_FOTA_DOWNLOAD_SECURITY_TAG != -1)
```

Reply:
If `IS_ENABLED` can be used instead of the preprocessor statement it would be better IMO. Preprocessor statements is not particulary readable.

Reply:
Using preprocessor statements is the way this is currently achieved, so I reused that part.
`IS_ENABLED` can - as far as I understand - not be used, since this is not a boolean value but the integer for the security tag?